### ISSUE TITLE
Discovery for BDD - SHA Part 1

### ISSUE THREAD

**Issue by RakshindaAslam** [department-of-veterans-affairs/va.gov-team#56573]
## Issue Description
Users who are active service members submitting a 526 under the BDD (Benefits Delivery at Discharge) Program will need to also submit a Part A Self-Assessment form as of April 1 in order to get an exam before discharge from service.

This ticket is to discover how we would make the front-end changes

Link to sketch file with URLs, current screenshots, and new mockups/wireframes showing changes *with red arrows*: https://sketch.com/s/23f30f99-cd9b-47c6-be44-e05a56f5d1b4

---
## Tasks
- [x] learn about front-end toggles and whether we could utilize them for this effort
- [x] learn how to make dynamic form changes and add relevant components, e.g. alert, checkbox
- [x] learn about relevant unit and e2e changes we will need
- [x] support the collab cycle by attending meeting(s), investigating and answering any technical questions 

## Acceptance Criteria
- [x] decide on toggle and deployment strategy
- [x] notes and/or branch with changes

---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [x] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `Console-Services`, `tools-fe`)
- [x] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [x] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

**Comment by RakshindaAslam** [comment]
cc @lydiahooper @christinec-fftc

**Comment by RakshindaAslam** [comment]
Pulled out of scope pending discovery - Design changes to Original Claim post-submission page

**Comment by lydiahooper** [comment]
All links in text should go to: https://www.benefits.va.gov/compensation/dbq_publicdbqs.asp

**Comment by christinec-fftc** [comment]
We initially had this BDD work broken out into two tickets (this ticket was design changes and #56570 for content changes), but after getting feedback from the design intent meeting and learning about front end toggles, we have decided to slice this up in a different way.
1 - discovery ticket (there is engineering work that can be done, but we can't commit to finishing the implementation until the design intent items have been resolved)
2 - implementation ticket

Will rework this ticket to be the discovery ticket and then update #56570 to be implementation

**Comment by christinec-fftc** [comment]
**Summary of Discovery findings**

**1. learn about front-end toggles and whether we could utilize them for this effort** ✅

Learned that a staging review will be required for these changes. Given the automated daily prod deployments, a toggle will be essential for the ability to merge the code changes so they are available on staging, but are suppressed for prod until the changes are approved. 

In terms of Front end toggles, there are a few different options
Using a toggle selector ([example](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/disability-benefits/all-claims/utils/index.jsx#L703))
Toggler component or react hook ([notes](https://dsva.slack.com/archives/C5HP4GN3F/p1680712065373399))
One of the above combined with session storage (this can reduce API calls, allow the config to be available throughout the user’s session, i.e. beyond the redux store, and reduce visible loading states for end users)

There is also another option to use environment variables ([example](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/disability-benefits/686c-674/config/chapters/household-income/householdIncome.js#L10)), however, it is not as feature-rich as using a toggle and would then require another code change + deployment to enable the feature in prod rather than setting the toggle.

Note that there are currently 5 different views where we would conditionally want to change the UI, so we should strongly consider removing all the toggle logic once the feature has settled in prod. This will keep the overall code more manageable over time and less spaghetti-fied.

One more caveat is that toggles are great to use in react components, but [can be challenging when trying to use within the form library configuration](https://dsva.slack.com/archives/CBU0KDSB1/p1607958122212900), i.e. RJSF.  An example of this for 526 is [the ticket for first pass of 526 paper/online sync](https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/56920) where we have to reorder views and move them into different chapters. In this case, we could possibly include them twice and use a toggle to conditionally display.

**2.  learn how to make dynamic form changes and add relevant components, e.g. alerts** ✅
I was able to make several of the proposed changes locally including 
Update the Introduction Page only if the form is for BDD and a toggle is enabled
Conditionally add a new alert based on user input and env check (we should be able to replace this with toggle check instead)
Text updates which should also be able to be gated by toggle or env check

I did not create an actual toggle yet, just used an existing one to make sure the conditionals were working

Note ui elements appearing dynamically after user input can be an a11y issue. There’s a ui prop option to alert screen readers that there’s new content

**3.  learn about relevant unit and e2e changes we will need** ✅
Unit tests
Snapshot tests are rarely used in the vets-website repo and are not being used for BDEx
Add tests to check for the presence of 3 new alert boxes when appropriate. 
Note this can be tricky with CI since CI runs in a prod-like state. Look for examples of adding a flag to override this.
Also add a test to verify alert does not display appropriately

E2E tests
bdd-claims.cypress.spec.js: since we are not adding or removing form inputs, no impact to e2e tests. They are running as is.
Since we are linking to a new form, we may need to add a new [attachment id](https://github.com/department-of-veterans-affairs/vets-json-schema/blob/master/dist/21-526EZ-ALLCLAIMS-schema.json#L1036) to vets-json-schema (when forms are uploaded they should have an id associated with them to help categorize the documents). We should be able to ask our intake person if they can advise if the SHA form should have a new document id associated with it or falls under ‘Other Correspondence’


**4.  support the collab cycle by attending meeting(s), investigating and answering any technical questions** ✅

**Comment by christinec-fftc** [comment]
Ignore my comment about the attachment id. I see the mocks have user instructions to use DBQ

**Comment by christinec-fftc** [comment]
Ignore my comment about the attachment id. I see the mocks have user instructions to use DBQ
